### DISCUSSION TITLE
Ldaprecord ActiveDirectory User does not have method can(..) using Gates

### DISCUSSION THREAD

**Discussion by wewhite** [DirectoryTree/LdapRecord#599] [Q&A] [answered]
Ldaprecord works, and I have middleware routes with gates working, using Laravel 10.

I now need to access authenticated users can method, but the method does not exist, because ```\LdapRecord\Models\ActiveDirectory\User``` class does not ```use Authorizable; ```

I have created my own User model that extends the ActiveDirectory User:
```
namespace App\Ldap;

use Illuminate\Foundation\Auth\Access\Authorizable;

class User extends \LdapRecord\Models\ActiveDirectory\User
{

    // need to use for $request->auth()->user()->can(...)
    use Authorizable;    
}
```
This enables me to now check in the Controller:
```
        $readonly = true;
        if ($request->user()->can('accessDashboard')) {
            $readonly = false;
        }
```

Is there a reason Authorizable is not part of LdapRecord user models?

**Comment by stevebauman** [comment, marked as answer]
Hi there @wewhite,

The core LdapRecord package is usable outside of the Laravel framework, so it is not possible to use this trait on its included models.
